tests/std/tests/P2474R2_views_repeat/test.cpp

```diff
@@ -0,0 +1,295 @@
+    } else if constexpr (_Signed_integer_like<B>) {
+        static_assert(same_as<ranges::range_difference_t<R>, B>);
+    } else {
+        static_assert(same_as<ranges::range_difference_t<R>, ranges::_Iota_diff_t<B>>);
+    }
+
+    static_assert(ranges::common_range<const R> == bounded);
+    static_assert(same_as<ranges::iterator_t<const R>, ranges::iterator_t<R>>);
+
+    static_assert(same_as<ranges::range_difference_t<const R>, ranges::range_difference_t<R>>);
+
+    static_assert(CanSize<R> == bounded);
+
+    same_as<R> auto rng = views::repeat(val, bound);
+    static_assert(noexcept(views::repeat(val, bound)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(rng | views::drop(1)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(rng | views::take(1)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(move(rng) | views::drop(1)) == is_nothrow_move_constructible_v<T>); // strengthened
+    static_assert(noexcept(move(rng) | views::take(1)) == is_nothrow_move_constructible_v<T>); // strengthened
+
+    if constexpr (bounded) {
+        B i = 0;
+        for (const auto& v : rng) {
+            assert(v == val);
+            ++i;
+        }
+        assert(i == bound);
+
+        static_assert(noexcept(rng.end())); // strengthened
+        assert(cmp_equal(rng.size(), bound));
+        static_assert(noexcept(rng.size())); // strengthened
+
+        constexpr int amount   = 3;
+        const auto size        = ranges::distance(rng);
+        const auto take_amount = ranges::min(size, amount);
+
+        const same_as<R> auto take = rng | views::take(amount);
+        assert(cmp_equal(take.size(), take_amount));
+
+        const auto drop_amount     = size - ranges::min(size, amount);
+        const same_as<R> auto drop = rng | views::drop(amount);
+        assert(cmp_equal(drop.size(), drop_amount));
+    } else {
+        static_assert(noexcept(rng.end()));
+
+        using ReconR = ranges::repeat_view<T, ranges::range_difference_t<R>>;
+
+        const same_as<ReconR> auto take = rng | views::take(3);
+        assert(take.size() == 3);
+
+        const same_as<R> auto drop = rng | views::drop(3);
+        static_assert(!CanSize<decltype(drop)>);
+    }
+
+    static_assert(noexcept(rng.begin())); // strengthened
+    using I = ranges::iterator_t<R>;
+    static_assert(same_as<typename I::iterator_concept, random_access_iterator_tag>);
+    static_assert(same_as<typename I::iterator_category, random_access_iterator_tag>);
+
+    static_assert(is_nothrow_default_constructible_v<I>);
+    assert(I{} == I{});
+    assert(!(I{} != I{}));
+    assert(!(I{} < I{}));
+    assert(!(I{} > I{}));
+    assert(I{} <= I{});
+    assert(I{} >= I{});
+    assert(I{} <=> I{} == 0);
+    assert(I{} - I{} == 0);
+
+    const I first  = rng.begin();
+    const I second = ranges::next(first);

```

Comment:
UB if `bound == 0`. (Indicates missing test coverage.)

Reply:
Why would that be UB?

Reply:
My mistake, this range is unlike every other in the Standard Library: iterators past the end are perfectly ok and even dereferenceable.